test/Psr/CacheItemPool/CacheItemTest.php

```diff
@@ -106,4 +110,32 @@ public function testExpiresAfterInvalidThrowsException(): void
         $item = new CacheItem('key', 'value', true);
         $item->expiresAfter([]);
     }
+
+    public function testExpiresAfterStartsExpiringAfterMethodCall(): void
+    {
+        $item = new CacheItem('key', 'value', false);
+        $item = $item->expiresAfter(10);
+        sleep(1);
+        self::assertEquals(9, $item->getTtl());
+    }
+
+    public function testClockProvidedDoesNotContainUTCTimeZone(): void
+    {
+        $item = new CacheItem(
+            'foo',
+            null,
+            false,
+            new SystemClock(new DateTimeZone('Europe/Berlin'))
+        );
+
+        $interval = DateInterval::createFromDateString('1 hour');
+        $item->expiresAfter($interval);
+
+        self::assertEquals(3600, $item->getTtl());
+
+        $inOneHour = (new DateTimeImmutable('now', new DateTimeZone('America/Vancouver')))->add($interval);
+        $item->expiresAt($inOneHour);
+
+        self::assertEquals(3600, $item->getTtl());
```

Comment:
This test is a bit fragile: may fail if `SystemClock` moves while we're observing this (for whatever reason, the test has a hiccup)

Reply:
I am open for another solution, I spent quite a while today and don't come to a good conclusion tho. Maybe tomorrow.

Reply:
You fixed this 👍